nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/hint/MoreLikeThisESHintQueryBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.elasticsearch.hint;
+
+import static org.nuxeo.elasticsearch.ElasticSearchConstants.DOC_TYPE;
+
+import org.elasticsearch.index.query.MoreLikeThisQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.nuxeo.ecm.core.api.repository.RepositoryManager;
+import org.nuxeo.ecm.core.query.sql.model.EsHint;
+import org.nuxeo.elasticsearch.api.ESHintQueryBuilder;
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 11.1
+ */
+public class MoreLikeThisESHintQueryBuilder implements ESHintQueryBuilder {
+
+    public static final int MORE_LIKE_THIS_MIN_TERM_FREQ = 1;
+
+    public static final int MORE_LIKE_THIS_MIN_DOC_FREQ = 3;
+
+    public static final int MORE_LIKE_THIS_MAX_QUERY_TERMS = 12;
+
+    @Override
+    public QueryBuilder make(EsHint hint, String fieldName, Object value) {
+        String[] indexFields = hint.getIndexFieldNames();
+        String[] fields = indexFields.length > 0 ? indexFields : new String[] { fieldName };
+        MoreLikeThisQueryBuilder moreLikeThisBuilder = QueryBuilders.moreLikeThisQuery(fields, null, getItems(value));
+        if (hint.analyzer != null) {
+            moreLikeThisBuilder.analyzer(hint.analyzer);
+        }
+        moreLikeThisBuilder.minDocFreq(MORE_LIKE_THIS_MIN_DOC_FREQ);
+        moreLikeThisBuilder.minTermFreq(MORE_LIKE_THIS_MIN_TERM_FREQ);
+        moreLikeThisBuilder.maxQueryTerms(MORE_LIKE_THIS_MAX_QUERY_TERMS);
+        return moreLikeThisBuilder;
+    }
+
+    public static MoreLikeThisQueryBuilder.Item[] getItems(Object value) {
+        RepositoryManager rm = Framework.getService(RepositoryManager.class);
+        String repo = rm.getDefaultRepository().getName();
+        ElasticSearchAdmin esa = Framework.getService(ElasticSearchAdmin.class);
+        String esIndex = esa.getIndexNameForRepository(repo);
+        String[] values;
+        if (value instanceof Object[]) {
+            values = (String[]) value;
```

Comment:
This is not consistent, if you want to cast to `String[]` you should do `instanceof String[]`

Reply:
I agree, the code was as this, in the original class and i didn't pay attention of this inconsistency.

Reply:
Hm ok if it's from original code that you're just moving to a separate class then ignore my comments, it should be cleaned up in a separate commit.

Reply:
i will do it, it's more clear and clean :)